src/rules.jl

```diff
@@ -135,6 +136,51 @@ function Base.show(io::IO, o::RMSProp)
     print(io, "; centred = ", o.centred, ")")
 end
 
+
+"""
+    Rprop(η = 1f-3, ℓ = (5f-1, 1.2f0), Γ = (1f-6, 50f0))
+
+Optimizer using the
+[Rprop](https://ieeexplore.ieee.org/document/298623) algorithm. An full-batch
+learning algorithm that depends only on the sign of the gradient.
+
+# Parameters
+- Learning rate (`η`): Amount by which gradients are discounted before updating
+                       the weights.
+                      
+- Scaling factors (`ℓ::Tuple`): Multiplicative increase and decrease factors.
+
+- Step sizes (`Γ::Tuple`): Mminimal and maximal allowed step sizes.
+"""
+struct Rprop{T} <: AbstractRule
+    eta::T
+    ell::Tuple{T,T}
+    gamma::Tuple{T,T}
+end
+
+Rprop(η = 1f-3, ℓ = (5f-1, 1.2f0), Γ = (1f-6, 50f0)) = Rprop{typeof(η)}(η, ℓ, Γ)
+
+init(o::Rprop, x::AbstractArray) = (zero(x), onevalue(o.eta, x))
+
+function apply!(o::Rprop, state, x, dx)
+    ℓ, Γ = o.ell, o.gamma
+    g₀, η₀ = state
+
+    signs = g₀ .* dx
+    signs[signs .> 0] .= ℓ[2]
+    signs[signs .< 0] .= ℓ[1]
+    signs[signs .== 0] .= one(eltype(signs))
+
+    η₁ =  clamp.(η₀ .* signs, Γ[1], Γ[2])
+
+    g₁ = copy(dx)
+    g₁[signs .== ℓ[1]] .= zero(eltype(g₁))
```

Comment:
I think you can probably write this as follows:
```suggestion
    @.. g = (signs == ℓ[1]) * dx
```
First, this avoids making and then mutating `g₁`, since it might be an immutable array. (For now, tested with StaticArrays.) 

Second, less importantly, `@..` writes back into the same `g` if it is in fact mutable, otherwise binds `g` to the RHS.